alacritty/src/renderer/rects.rs

```diff
@@ -399,44 +402,47 @@ pub struct RectShaderProgram {
     /// Shader program.
     program: ShaderProgram,
 
-    /// Kind of rect we're drawing.
-    u_rect_kind: GLint,
-
     /// Cell width.
-    u_cell_width: GLint,
+    u_cell_width: Option<GLint>,
```

Comment:
This isn't actually how this struct works, right? But instead we have a very specific set of available combinations that are allowed?

We might want to make this a `union` instead?

Reply:
I don't think this is clear at all though, is it?

You could easily create an invalid struct, could you not? There's only a very limited amount of combinations that are actually "legal" and this struct does nothing to prevent the construction from illegal variants?

Reply:
By unions you mean `enum`? Since I won't use C union for that.